sdks/typescript/src/clients/listeners/durable-listener/pooled-durable-listener-client.ts

```diff
@@ -0,0 +1,227 @@
+// eslint-disable-next-line max-classes-per-file
+import { EventEmitter, on } from 'events';
+import {
+  DurableEvent,
+  ListenForDurableEventRequest,
+  RegisterDurableEventRequest,
+  RegisterDurableEventResponse,
+} from '@hatchet/protoc/v1/dispatcher';
+import { isAbortError } from 'abort-controller-x';
+import sleep from '@hatchet/util/sleep';
+import {
+  DurableEventListenerConditions,
+  SleepMatchCondition,
+  UserEventMatchCondition,
+} from '@hatchet/protoc/v1/shared/condition';
+import { DurableListenerClient } from './durable-listener-client';
+
+export class DurableEventStreamable {
+  listener: AsyncIterable<DurableEvent>;
+  taskId: string;
+  signalKey: string;
+
+  responseEmitter = new EventEmitter();
+
+  constructor(listener: AsyncIterable<DurableEvent>, taskId: string, signalKey: string) {
+    this.listener = listener;
+    this.taskId = taskId;
+    this.signalKey = signalKey;
+  }
+
+  async get(): Promise<DurableEvent> {
+    return new Promise((resolve) => {
+      this.responseEmitter.once('response', resolve);
+    });
+  }
+}
+
+export class DurableEventGrpcPooledListener {
+  listener: AsyncIterable<DurableEvent> | undefined;
+  requestEmitter = new EventEmitter();
+  signal: AbortController = new AbortController();
+  client: DurableListenerClient;
+
+  subscribers: Record<string, DurableEventStreamable> = {};
+  taskSignalKeyToSubscriptionIds: Record<string, string[]> = {};
+  onFinish: () => void = () => {};
+
+  private subscriptionCounter = 0;
+  private currRequester = 0;
+  private readonly DEFAULT_INTERRUPT_INTERVAL = 1800000; // 30 minutes in milliseconds
```

Comment:
nit: for this kind of thing, I've always liked: `30 * 60 * 1000`

Reply:
good catch